Sources/SwiftKuery/Index.swift

```diff
@@ -77,8 +73,14 @@ public struct Index {
     /// - Parameter connection: The connection to the database.
     /// - Parameter onCompletion: The function to be called when the execution of the query has completed.
     public func drop(connection: Connection, onCompletion: @escaping ((QueryResult) -> ())) {
-        var query = "DROP INDEX "
-        query += Utils.packName(name, queryBuilder: connection.queryBuilder)
+        let queryBuilder = connection.queryBuilder
+        var query: String
+        if queryBuilder.plugin == .mysql {
+            query = "ALTER TABLE \(Utils.packName(table._name, queryBuilder: queryBuilder)) DROP INDEX "
+        } else {
+            query = "DROP INDEX "
+        }
+        query += Utils.packName(name, queryBuilder: queryBuilder)
```

Comment:
Was finding it hard to plug this in a generic manner, so added an enum that can be used for other such customizations.